shell.nix

```diff
@@ -0,0 +1,39 @@
+with import (
+    # to ge the hash.
+builtins.fetchTarball {
+    # To update
+    #     curl -sI https://nixos.org/channels/nixpkgs-unstable/nixexprs.tar.xz | awk '/Location:/ {print $2}'
+    url = "https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-18.09pre146409.2391d1dbf31/nixexprs.tar.xz";
+    # To get the hash
+    #     nix-prefetch-url --unpack $url
+    sha256 = "08iy7dlyy5f7w1npajrb5pi7gl6mwfjimkkpbcqmkz01bam9bq5x";
+  }
+) { };
+
+# plyvl requires leveldb >= 1.20
+let myLeveldb = callPackage ./leveldb.nix {};
+
+in mkShell rec {
+
+  # Might be needed on mac OS? If so, consider setting on darwin automatically.
+  # LC_ALL="en_US.UTF-8";
+
+  # Cannot build wheel otherwise (zip 1980 issue).
+  SOURCE_DATE_EPOCH="315532800";
+
+  buildInputs = [
+    libffi
+    openssl
+    gmp
+    pkgconfig
+    python36
+    python36Packages.virtualenv
+    solc
+    myLeveldb
+    libxml2
+    libxslt
+    go-ethereum
```

Comment:
No. It will fetch the version of `geth` in the nix expressions specified on top and then install that into the nix store in `/nix/store`. Then only if the nix shell is activated it will use this version of `geth` because it's `bin` directory is pre-pended to the `PATH`. When deactivating the shell this would be removed so one can still use the system `geth`.

Inside the nix shell the PATH becomes something like

```
[nix-shell:~/dev/raiden]$ echo $PATH | tr ':' '\n'
/nix/store/nzdzpzxw41638d8lqxcqbdypmwvzd3ql-patchelf-0.9/bin
/nix/store/97vx23sqvi8jmnhpmr8rnpiwkjz08r55-paxctl-0.9/bin
/nix/store/38picpf5cl6d7lj38h8191vmxlxjbgq3-gcc-wrapper-7.3.0/bin
/nix/store/bkl2zbbvhq8ykn7536gi5y17q2sb0wnd-gcc-7.3.0/bin
/nix/store/syxvd6asylby0nxmcc1f3vf1f2l014if-glibc-2.27-bin/bin
/nix/store/l816bx8lgm0jwb7hzbc105h0fv097zry-coreutils-8.29/bin
/nix/store/f6gkar97rvp7kxvzr77g6qdjzl9frcvx-binutils-wrapper-2.30/bin
/nix/store/90vmpr41dzsx350k5argycaf693hnl1l-binutils-2.30/bin
/nix/store/syxvd6asylby0nxmcc1f3vf1f2l014if-glibc-2.27-bin/bin
/nix/store/l816bx8lgm0jwb7hzbc105h0fv097zry-coreutils-8.29/bin
/nix/store/nylyf9hmrl947zni8xv01smp2x39v0f5-openssl-1.0.2o-bin/bin
/nix/store/5r3lfq3qdwk96ahn2wcxh4pjs4qhih80-pkg-config-0.29.2/bin
/nix/store/ihxxqhpdx3v06la2h02mgqkfvfhhymv3-python3-3.6.6/bin
/nix/store/8ia8w3q1s5xgs16l714khwcm5pv8m06q-python3.6-virtualenv-16.0.0/bin
/nix/store/8zhvb6873v9x2905av41d9q8cih6gpas-python3.6-setuptools-39.2.0/bin
/nix/store/6fdk71zxws2ha6a7d3fvwwqngwkawdqm-solc-0.4.24/bin
/nix/store/vh062y13zyxpyklpwqpynhd4ddzqika2-libxml2-2.9.8-dev/bin
/nix/store/bhmvdjrir06l4akbd540il8679zba7l9-libxml2-2.9.8-bin/bin
/nix/store/p71r0adjq2knbz1a5ghzn75i2s69p8pz-libxslt-1.1.32-dev/bin
/nix/store/rbgbax009b0wkjihfh58fi5ahmybqsxv-libxslt-1.1.32-bin/bin
/nix/store/zr0pisgz07jd3kffqhgy2ab2076sk2b3-go-ethereum-1.8.11-bin/bin
/nix/store/4wzzgqhwp0hdxd6i38vsdpg64vczkafz-git-2.18.0/bin
/nix/store/l816bx8lgm0jwb7hzbc105h0fv097zry-coreutils-8.29/bin
/nix/store/8waljb40kcykpiq683nm5rfwhqsy3p92-findutils-4.6.0/bin
/nix/store/fci1flnnlr0244flannyimdn6wpp1398-diffutils-3.6/bin
/nix/store/p1hfw3s0ss3ak3fn20pa6974915vwpac-gnused-4.5/bin
/nix/store/9yrdb1ba4ha2clp9mh6cw6dzhyh9i6k6-gnugrep-3.1/bin
/nix/store/13ggwv7w8y9jaand0byk88x8fgfbjsal-gawk-4.2.1/bin
/nix/store/9mwqdnc7qfjj4i7ljv03jh96bcy89kr1-gnutar-1.30/bin
/nix/store/g2v7535bvv8rzzldfhic1ia551lh91pq-gzip-1.9/bin
/nix/store/scks4z2bpgvclcy562xab9k9ndxqxklm-bzip2-1.0.6.0.1-bin/bin
/nix/store/w93xhdza9g1kmkj93qr8llf260w7i4v7-gnumake-4.2.1/bin
/nix/store/z9qg7lpkrajdnlgw5pxp9yx984prl1fi-bash-4.4-p23/bin
/nix/store/j6f7wyag2ph93bx4wsjrm6mjrpcl16k3-patch-2.7.6/bin
/nix/store/kj2fggp62fj9cchp7y00x4znhj24g3mr-xz-5.2.4-bin/bin
/home/lulu/bin
/home/lulu/.local/bin
/home/lulu/go/bin
/home/lulu/.nix-profile/bin
/home/lulu/.local/bin
/home/lulu/.fzf/bin
/home/lulu/.nix-profile/bin
/home/lulu/.local/bin
/bin
/usr/bin
/usr/local/bin
/usr/local/sbin
/opt/cuda/bin
/usr/lib/jvm/default/bin
/usr/bin/site_perl
/usr/bin/vendor_perl
/usr/bin/core_perl
```

Reply:
We could also pin a certain version of `geth` if that would be desirable, now it is `1.8.11`.